### ISSUE TITLE
Change focus via keyboard

### ISSUE THREAD

**Issue by felagund** [Aegisub/Aegisub#106]
In the manual, it says that Aegisub relies heavily on the concept of focus. Remarkably, I seem to be unable to find away to change focus via keyboard.

Like, how would I go from video focus to edit the line that is selected in the subtitle grid and then go back?

**Comment by ilko-k** [comment]
assign the corresponding function to a key 👍 
"subtitle/select/visible"

then hit ctrl+P to resume playback (and regain focus on video at the same time)

**Comment by felagund** [comment]
Sorry. I am not sure if there is any user-support forum or something like that (i tried to find a way to reasonably search the archived forums on internet archive but failed).

Anyway, this does not ever removes the cursor from the text field with the subtitle text (I am unable to select subtitle grid, that is) (I have "focus grid on click" checked).

**Comment by felagund** [comment]
(if you want it to be a bug reporet, the link in help>forums lead to a error page on aegisub.org :-D)

**Comment by ilko-k** [comment]
oh, yeah I see but still, it's not an issue. Aegisub has 3 areas where you can work with hotkeys. You can't even switch with the tab key (I don't know why, but might be a mistake after all). I advise you to re-adapt your methods of work. I do fansubbing since a few years and with my actual configuration, I don't need to focus on the subtitle grid or the video area unless for very specific jobs (drawing, moves...).
"default hotkeys" override all other hotkeys whatever the focus. It's up to you to turn this feature into a solution instead of considering it as an obstacle.

**Comment by felagund** [comment]
Well, I do not do funsubbing but gte actually paid for it, but I guess that does not matter much, Aegisub still comes useful for simple text efects.

I actually need to focus on the video area all the time. Meaning I need to be able to use shortcuts to move around the video, use one file shortcuts to do that preferably and so on. I usually have to time the subtitles to the video, not audio (I would still use Subtitle Editor for that, it is much cleaner) - and it needs to be frame-precise.  And I cannot do that while editing subtitle text. But I would need to switch between the too - and I have to use the mouse to do it now, which, if it is not possible, I think must be a bug.

**Comment by felagund** [comment]
Hm, so I actually tried to implement this in my setup. Can I ever focus the selected line so that I can edit it via a keyboard? "subtitle/select/visible" does not seem to do that, either from video or from the subtitle grid. Also, when I use "video/focus_seek" when editing a line, it duly toggles focus to the video slider, but never goes back to the text area. Like is there any way to focus the area in order to edit the text without using a mouse? No matter what my workflow is, this just should be possible, should it not?

**Comment by ilko-k** [comment]
"subtitle/select/visible" is only possible when the play position matches the corresponding line(s). This will stop playback and focus on them. But TBH, I don't get why you are persisting about the video. You can work accurately on frames with hotkeys without losing focus on the text area.

**Comment by felagund** [comment]
I see. Setting video/frame/next and prev under Default indeed does this. But if your suggestion is to put all shortcuts in default and never change focus, than I am not sure why the manual says that Aegisub relies heavily on the concept of focus when I am never supposed to change focus.

My main motivation was to use single key shortcuts, which I cannot when editing a line. I mean, I can reassign the function keys, but that is not necessarily the most ergonomic (and I might use the numpad maybe - it is true that since I first asked, I started to use an external keyboard that has it).

Also, after using a video tool like dragging a subtitle (which focuses video), sometimes I need to edit it, change its size etc. - a way to refocus editing a line would we welcome. 

Another motivation is that when editing a line, any mistake I make (like pressing a wrong shortcut, that might not have anythign assigned for example) will result into editing that line and I have to delete it. When focusing the video, such a command would just silently fail.

**Comment by ilko-k** [comment]
hmm, maybe there is a misunderstanding of the term "focus" here. In case of Aegisub, it's meant to work on an area so that you don't need to worry about anything else while you're "focused" on it. As specified in the documentation, this implies to do several "passes".

What kind of modifications are you talking about that need to focus on the grid ? Just curious.

**Comment by felagund** [comment]
I am not sure Aegisub enforces (or should enforce) such usage. I do not really use it for timing (I think there are better programs for that), but it is excellent for effects, so technically I do two "passes". But even then I need to change focus frequently. Let's say I am trying to translate a text sign. I might want to change the translation in order to fit it on the screen or just change line breaks. Even more importatntly, I enter some text effects (like font size) directly via keyboard as code in the text box and some graphically with a mouse when video is focused. I am not really sure how I could do such a thing in a single pass without changing focus. Ergo, I am not sure how wanting to be able to change focus via keyboard goes agains Aegisub philosophy.

As for grid, I do not think I need to focus it as such (using time/next and time/prev I can already move between subtitles) but I sometimes focus it when I press arrow key down to move between the subtitles, which even after years still feel natural.

But what i would need is an action to focus the text field of a subtitle selected in the subtitle grid and then move back so that video is focus (=there is no cursor in the text field).

**Comment by ilko-k** [comment]
yeah the TAB key could be used for this purpose. Something like CTRL+TAB, or even the function keys.

**Comment by felagund** [comment]
So this can stay here as an feature request, though if I understand it correctly, there is nobody actively developing Aegisub, is there?
